Source/CarthageKit/VersionFile.swift

```diff
@@ -237,6 +237,72 @@ struct VersionFile: Codable {
 	}
 }
 
+/// Creates a version file for the current project in the
+/// Carthage/Build directory which associates its commitish with
+/// the hashes (e.g. SHA256) of the built frameworks for each platform
+/// in order to allow those frameworks to be skipped in future builds.
+///
+/// Derives the current project name from `git remote get-url origin`
+///
+/// Returns a signal that succeeds once the file has been created.
+public func createVersionFileForCurrentProject(
+	platforms: Set<Platform>,
+	buildProducts: [URL],
+	rootDirectoryURL: URL
+) -> SignalProducer<(), CarthageError> {
+
+	// List all remotes known for this repository
+	let allRemotes = launchGitTask(["remote"])
+		.flatMap(.concat) { $0.linesProducer }
+		.map { $0.trimmingCharacters(in: .whitespacesAndNewlines) }
+
+	// For each remote, get the `get-url`
+	let allRemoteURLs = allRemotes
+		.flatMap(.concat) { launchGitTask(["remote", "get-url", $0]) }
+		.map { $0.trimmingCharacters(in: .whitespacesAndNewlines) }
+		.collect()
+
+	let currentProjectAsDependecy = allRemoteURLs
+		// Assess the popularity of each remote url
+		.map { $0.reduce([String: Int]()) { map, remoteURL in
+			var remoteURLPopularityMap = map
+			remoteURLPopularityMap[remoteURL] = (map[remoteURL] ?? 0) + 1
+			return remoteURLPopularityMap
+			}
+		}
+		// Sort remotes by popularity or alphabetically in case of a draw
+		.map { (remoteURLPopularityMap: [String: Int]) -> (url: String, popularity: Int) in
+			remoteURLPopularityMap.sorted { lhs, rhs in
+				if lhs.value == rhs.value {
+					return lhs.key < rhs.key
+				}
+				return lhs.value > rhs.value
+			}
+		    // This should never happen... but use a fallback anyways
+			.first ?? ("_Current", 0)
+		}
```

Comment:
I think it'd be nice to default to `origin` if it exists.

* Most people will only have 1 remote.
* Most people with >1 remote will have an `origin`.
* If you don't have either, who knows what you'll have?

But choosing `origin` over the "popular" or alphabetical option seems like a better choice.

Reply:
ok

Reply:
Done